src/main/java/com/philips/research/bombase/controller/LicensesRoute.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2020-2021, Koninklijke Philips N.V., https://www.philips.com
+ * SPDX-License-Identifier: MIT
+ */
+
+package com.philips.research.bombase.controller;
+
+import com.philips.research.bombase.core.license_cleaner.LicenseCleanerService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import pl.tlinkowski.annotation.basic.NullOr;
+
+@RestController
+@CrossOrigin(origins = "*")
+@RequestMapping("/licenses")
+public class LicensesRoute {
+    private final LicenseCleanerService service;
+
+    LicensesRoute(LicenseCleanerService service) {
+        this.service = service;
+    }
+
+    @PostMapping()
+    void setLicenseCuration(@RequestBody CurationJson body) {
+        if (body.license == null || body.curation == null) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Need license and curation in body");
+        }
+        service.defineCuration(body.license, body.curation);
+    }
+
+    static class CurationJson {
```

Comment:
Why does this need to be a static class?

Reply:
The difference is that a "normal" class inherits the context of its parent class, so it can only be created "inside" a parent class. (Apparently) Jackson can work around this by implicitly creating a parent class, but that takes a bit of extra memory.

Therefore it is a good habit to make inner classes that do not refer to their parent `static`.

Reply:
Thanks for the explanation.